odin/client/main.go

```diff
@@ -0,0 +1,211 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	app "github.com/GeoDB-Limited/odin-core/app"
+	odinapp "github.com/GeoDB-Limited/odin-core/app"
+	odinminttypes "github.com/GeoDB-Limited/odin-core/x/mint/types"
+	"github.com/GeoDB-Limited/odin-deposit-ether-svc/internal/config"
+	sdktxclient "github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	sdkauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	sdkauthtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"io/ioutil"
+	"math/big"
+)
+
+var encoding = odinapp.MakeEncodingConfig()
+
+// Client defines an interface for the wrapped cosmos sdk service client.
+type Client interface {
+	WithSigner()
+	GetAccount(string) (sdkauthtypes.AccountI, error)
+	SetBridgeAddress(common.Address) error
+	GetBridgeAddress() (common.Address, error)
+	ClaimWithdrawal(string, *big.Int) error
+	GetExchangeRate(string) (*big.Int, error)
+}
+
+// client defines typed wrapper for the cosmos sdk service client.
+type client struct {
+	connection *grpc.ClientConn
+	config     config.Config
+	signer     *signer
+}
+
+// signer defines data to sign the transactions
+type signer struct {
+	address    sdk.AccAddress
+	privateKey *secp256k1.PrivKey
+}
+
+// New creates a client that uses the given cosmos sdk service client.
+func New(cfg config.Config) Client {
+	odinConfig := cfg.OdinConfig()
+	conn, err := grpc.Dial(odinConfig.Endpoint, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		panic(errors.Wrapf(err, "failed to dial %s", odinConfig.Endpoint))
+	}
+
+	return &client{
+		connection: conn,
+		config:     cfg,
+	}
+}
+
+// WithSigner initializes odin signer to broadcast transactions
+func (c *client) WithSigner() {
```

Comment:
It is unclear that this method updates the state, make it return the updated object and do not modify the passed one